### PR TITLE
feat: invisible sync hooks (ADR-066)

### LINKED ISSUE #825: feat: post-merge hook auto-syncs when lessons change
When `git pull` triggers a merge, detect if `.totem/lessons/` changed via `git diff-tree`. If so, run `totem sync --quiet` in the background. Expand the existing `installPostMergeHook` in `install-hooks.ts`. See ADR-066.

### LINKED ISSUE #826: feat: post-checkout hook syncs context on branch switch
When switching branches, detect if `.totem/` directory differs between branches. If so, trigger background sync so the vector index matches the current branch's rules. New hook script alongside existing pre-push and post-merge. See ADR-066.

### LINKED ISSUE #827: feat: totem sync --quiet flag for background/hook usage
Add `--quiet` flag that suppresses all stdout output. Required for Git hook integration where sync runs in the background. `totem sync` currently logs progress which interrupts terminal flow during `git pull/checkout`. See ADR-066.

### PR DESCRIPTION
**Pull request by satur8d**
## Summary
- **#825** — Post-merge hook only triggers background sync when `.totem/lessons/` files actually changed (via `git diff-tree`). Adds `--quiet` flag to hook command and deterministic `# [totem] end` markers for safe eject scrubbing.
- **#827** — Implements `-q, --quiet` flag on `totem sync` command that suppresses spinner/stdout for background hook usage. Fatal errors still propagate via stderr.
- **#826** — New post-checkout hook triggers background sync when `.totem/` files differ between branches. Guards file checkouts (`$3=0`) and handles initial clone null-SHA edge case.

All three complete the **ADR-066: Invisible Sync Hooks** initiative — Git lifecycle events now keep the vector index fresh without developer intervention.

Closes #825
Closes #826
Closes #827

## Test plan
- [ ] 665 CLI tests pass (11 new across install-hooks, eject, sync)
- [ ] `totem lint` — 0 errors, 219 rules
- [ ] Pre-push gate passes (format + lint + full test suite)
- [ ] Verify `totem hooks` installs 4 hooks (pre-commit, pre-push, post-merge, post-checkout)
- [ ] Verify `totem eject --force` cleanly removes both post-merge and post-checkout hooks
- [ ] Verify `totem sync --quiet` produces no stdout

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added --quiet / -q flag to totem sync to suppress progress spinner
  * Install now includes a post-checkout git hook to run background sync on branch switches

* **Bug Fixes**
  * Eject now more reliably cleans Totem hook blocks while preserving user content and records any skipped/failed removals

* **Tests**
  * Expanded unit tests for hook installation, scrub/removal logic, and sync command spinner behavior

### CODE REVIEW

File: packages/cli/src/commands/install-hooks.ts [hunk 1]
```diff
@@ -20,15 +23,45 @@ export function detectTotemPrefix(cwd: string): string {
...
 # ${TOTEM_HOOK_MARKER} — background re-index after pull/merge.
 
-echo "[totem] Triggering background re-index..."
-(${syncCmd} > .git/totem-sync.log 2>&1) &
+# Only sync when lessons changed (suppress errors if ORIG_HEAD is missing)
+if git diff-tree -r --name-only ORIG_HEAD HEAD 2>/dev/null | grep -q '\\.totem/lessons/'; then
+  echo "[totem] Lessons changed — triggering background re-index..."
+  (${syncCmd} > .git/totem-sync.log 2>&1) &
+fi
+# ${TOTEM_HOOK_END}
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/commands/install-hooks.ts]: Intentionally keeping `2>&1` — this is a **background** process spawned during `git pull`/`checkout`. If stderr is not redirected, Node.js stack traces would splatter across the terminal mid-git-operation. The log file (`.git/totem-sync.log`) exists specifically for debugging hook failures after the fact.

The `--quiet` flag handles the Node.js layer (suppress spinner/stdout); the shell redirect handles the OS layer (contain all background output). Both are intentional.

The echo was removed in 59538d2 — hooks are now fully silent on success.
